pkg/kubelet/pod_workers.go

```diff
@@ -236,3 +278,54 @@ func (p *podWorkers) checkForUpdates(uid types.UID) {
 		p.isWorking[uid] = false
 	}
 }
+
+// killPodNow returns a KillPodFunc that can be used to kill a pod.
+// It is intended to be injected into other modules that need to kill a pod.
+func killPodNow(podWorkers PodWorkers) eviction.KillPodFunc {
+	return func(pod *api.Pod, status api.PodStatus, gracePeriodOverride *int64) error {
+		// we timeout and return an error if we dont get a callback within a reasonable time.
+		gracePeriod := int64(0)
+		if gracePeriodOverride != nil {
+			gracePeriod = *gracePeriodOverride
+		} else if pod.Spec.TerminationGracePeriodSeconds != nil {
+			gracePeriod = *pod.Spec.TerminationGracePeriodSeconds
+		}
+
+		// the default timeout is relative to the grace period (we settle on 2s to wait for kubelet->runtime traffic to complete in sigkill)
+		timeout := int64(gracePeriod + (gracePeriod / 2))
+		minTimeout := int64(2)
+		if timeout < minTimeout {
+			timeout = minTimeout
+		}
+		timeoutDuration := time.Duration(timeout) * time.Second
+		timeoutTicker := time.NewTicker(timeoutDuration)
+		defer timeoutTicker.Stop()
+
+		// open a channel we block against until we get a result
+		type response struct {
+			err error
+		}
+		ch := make(chan response)
+		podWorkers.UpdatePod(&UpdatePodOptions{
+			Pod:        pod,
+			UpdateType: kubetypes.SyncPodKill,
+			OnCompleteFunc: func(err error) {
+				ch <- response{err: err}
+			},
+			KillPodOptions: &KillPodOptions{
+				PodStatusFunc: func(p *api.Pod, podStatus *kubecontainer.PodStatus) api.PodStatus {
+					return status
+				},
+				PodTerminationGracePeriodSecondsOverride: gracePeriodOverride,
+			},
+		})
+
+		// wait for either a response, or a timeout
+		select {
+		case r := <-ch:
+			return r.err
+		case <-timeoutTicker.C:
```

Comment:
nit: use `time.After()` instead so that you don't have to call `timeoutTicker.Stop`